File: livingrimoire start here/LivinGrimoire java/src/Auxiliary_Modules/PhraseInflector.java
package Auxiliary_Modules;

import java.util.HashMap;
import java.util.Map;

public class PhraseInflector {

    // Maps for pronoun and verb inflection
    private static final Map<String, String> inflectionMap = new HashMap<>();

    static {
        // Pronoun mappings
        inflectionMap.put("i", "you");
        inflectionMap.put("me", "you");
        inflectionMap.put("my", "your");
        inflectionMap.put("mine", "yours");
        inflectionMap.put("you", "i"); // Default inflection
        inflectionMap.put("your", "my");
        inflectionMap.put("yours", "mine");

        // Verb conjugations
        inflectionMap.put("am", "are");
        inflectionMap.put("are", "am");
        inflectionMap.put("was", "were");
        inflectionMap.put("were", "was");

        // Contractions
        inflectionMap.put("i'd", "you would");
        inflectionMap.put("i've", "you have");
        inflectionMap.put("you've", "I have");
        inflectionMap.put("you'll", "I will");
    }

    // Function to inflect a phrase
    public static String inflectPhrase(String phrase) {
        String[] words = phrase.split(" ");
        StringBuilder result = new StringBuilder();

        for (int i = 0; i < words.length; i++) {
            String word = words[i];
            String lowerWord = word.toLowerCase();
            String inflectedWord = word; // Default to the original word

            // Check if the word needs to be inflected
            if (inflectionMap.containsKey(lowerWord)) {
                inflectedWord = inflectionMap.get(lowerWord);

                // Special case for "you"
                if (lowerWord.equals("you")) {
                    // Inflect to "me" if it's at the end of the sentence or after a verb
                    if (i == words.length - 1 || (i > 0 && isVerb(words[i - 1].toLowerCase()))) {
                        inflectedWord = "me";
                    } else {
                        inflectedWord = "I";
                    }
                }
            }

            // Preserve capitalization
            if (Character.isUpperCase(word.charAt(0))) {
                inflectedWord = inflectedWord.substring(0, 1).toUpperCase() + inflectedWord.substring(1);
            }

            result.append(inflectedWord).append(" ");
        }

        return result.toString().trim();
    }

    // Helper function to check if a word is a verb
    private static boolean isVerb(String word) {
        return word.equals("am") || word.equals("are") || word.equals("was") || word.equals("were") ||
                word.equals("have") || word.equals("has") || word.equals("had") ||
                word.equals("do") || word.equals("does") || word.equals("did");
    }
}

File: livingrimoire start here/LivinGrimoire Kotlin/kotlin/auxiliary_modules/PhraseInflector.kt
package auxiliary_modules

import java.util.*

object PhraseInflector {
    // Maps for pronoun and verb inflection
    private val inflectionMap: MutableMap<String, String> = HashMap()

    init {
        // Pronoun mappings
        inflectionMap["i"] = "you"
        inflectionMap["me"] = "you"
        inflectionMap["my"] = "your"
        inflectionMap["mine"] = "yours"
        inflectionMap["you"] = "i" // Default inflection
        inflectionMap["your"] = "my"
        inflectionMap["yours"] = "mine"

        // Verb conjugations
        inflectionMap["am"] = "are"
        inflectionMap["are"] = "am"
        inflectionMap["was"] = "were"
        inflectionMap["were"] = "was"

        // Contractions
        inflectionMap["i'd"] = "you would"
        inflectionMap["i've"] = "you have"
        inflectionMap["you've"] = "I have"
        inflectionMap["you'll"] = "I will"
    }

    // Function to inflect a phrase
    fun inflectPhrase(phrase: String): String {
        val words = phrase.split(" ".toRegex()).dropLastWhile { it.isEmpty() }.toTypedArray()
        val result = StringBuilder()
        for (i in words.indices) {
            val word = words[i]
            val lowerWord = word.lowercase(Locale.getDefault())
            var inflectedWord: String? = word // Default to the original word

            // Check if the word needs to be inflected
            if (inflectionMap.containsKey(lowerWord)) {
                inflectedWord = inflectionMap[lowerWord]

                // Special case for "you"
                if (lowerWord == "you") {
                    // Inflect to "me" if it's at the end of the sentence or after a verb
                    inflectedWord =
                        if (i == words.size - 1 || i > 0 && isVerb(words[i - 1].lowercase(Locale.getDefault()))) {
                            "me"
                        } else {
                            "I"
                        }
                }
            }

            // Preserve capitalization
            if (Character.isUpperCase(word[0])) {
                inflectedWord =
                    inflectedWord!!.substring(0, 1).uppercase(Locale.getDefault()) + inflectedWord.substring(1)
            }
            result.append(inflectedWord).append(" ")
        }
        return result.toString().trim { it <= ' ' }
    }

    // Helper function to check if a word is a verb
    private fun isVerb(word: String): Boolean {
        return word == "am" || word == "are" || word == "was" || word == "were" || word == "have" || word == "has" || word == "had" || word == "do" || word == "does" || word == "did"
    }
}